### PR TITLE
Added: Garbage Collection & Deletion of Library Items

### PR DESCRIPTION
**Pull request by Sewer56**
This PR contains the lightweight Garbage Collection implementation.
As well as the ability to remove items from the library.

The Garbage Collector itself is fairly modular, with its core being abstracted away from the App. This should make long term maintenance easier as models, and other implementation details change. More specific code, like App DataModel, Nx etc. lives within specialized projects per category.

- Generic code lives `NexusMods.App.GarbageCollection`
- Nx specific code lives in `NexusMods.App.GarbageCollection.Nx`
- DataModel specific code lives in `NexusMods.App.GarbageCollection.DataModel`

This PR also fixes the documentation sidebar.

---------------

This PR is ready.

~~However expect the design for the warning dialog below to change~~

image

~~So it's therefore still marked as 'draft'.~~

~~Simply ignore, the XAML files when reviewing until finalized.~~

### CODE REVIEW

**Comment by Patriot99** [comment]
The current message, `Are you sure you want to delete these items?` is ambiguous when the user tries to delete only one item, as the word “these” suggests multiple items are being referred to. A better solution would be to use dynamic content that adjusts the message based on the number of items to be deleted. For instance, if the user wants to delete one item, the message could read: **Are you sure you want to delete this item?** If multiple items are being deleted, the message could remain: `Are you sure you want to delete these items?` Alternatively, a more general phrase such as "Are you sure you want to delete the selected items?" would be appropriate for both single and multiple item deletions. It is important to make the message clear and unambiguous so that the user fully understands the consequences of confirming the action.

**Comment by Sewer56** [comment]
@Patriot99 This dialog box is a placeholder. I only got the proper design for this earlier today, so the text will change.

**Comment by Sewer56** [comment]
After doing a quick talk with @captainsandypants (CC. @Nexus-Mods/nexusmods-app-developers)

1. Please review this PR, just ignore the new dialog XAML, that's getting replaced.
2. We're only going to show the dialog when a user is going to delete a mod that's in a number of loadout(s) for now.
    - The code for the other possible warning conditions I've been requested to keep.
        - We'll show them in the future based on user feedback from next release(s).

I'm going to adapt the dialog to the design which is now in https://github.com/Nexus-Mods/NexusMods.App/issues/1899 , should be done by tomorrow.

File: tests/NexusMods.DataModel.Synchronizer.Tests/GeneralLoadoutManagementTests.SynchronizerIntegrationTests.verified.md [hunk 1]
```diff
@@ -114,7 +114,7 @@ A new loadout is created, but it has not been synchronized yet. So again the 'La
 ### Loadout B - (1)
 | Path | Hash | Size | TxId |
 | --- | --- | --- | --- |
-| (EId:20000000000000D, Game, bin/originalGameFile.txt) | 0xA52B286A3E7F4D91 | 12 B | Tx:10000000000000F |
+| (EId:20000000000000C, Game, bin/originalGameFile.txt) | 0xA52B286A3E7F4D91 | 12 B | Tx:10000000000000E |
```
Comment (halgari) [inline tests/NexusMods.DataModel.Synchronizer.Tests/GeneralLoadoutManagementTests.SynchronizerIntegrationTests.verified.md:117]: I'm not sure why these values changed, I won't hold up the PR over it, but I'm wondering what new info we've added/removed that makes this change

**Comment by halgari** [comment]
In the future I'd like this sort of PR broken into smaller steps of: base functionality (GC collection, repacking, etc), advanced functionality (async running, progress, etc.), and UI. Makes it eaiser to review that way. But it all looks good, and it's always great to see a lot of tests on a PR.

File: tests/NexusMods.DataModel.Synchronizer.Tests/GeneralLoadoutManagementTests.SynchronizerIntegrationTests.verified.md (hunk 1 above)
Comment (Sewer56) [reply tests/NexusMods.DataModel.Synchronizer.Tests/GeneralLoadoutManagementTests.SynchronizerIntegrationTests.verified.md:117]: One of the fixes I made back when working on the garbage collection part a while back was making sure we don't pack empty archives.

That could happen in some scenarios like managing a game during tests for example.
That caused EId to shift down by one, as we're no longer writing metadata for empty archives.

**Comment by Sewer56** [comment]
> In the future I'd like this sort of PR broken into smaller steps of: base functionality 

I did actually try to split it up this time, but no fish bit the line on the review part unfortunately. Originally this PR was https://github.com/Nexus-Mods/NexusMods.App/pull/1808 .

It was originally submitted as GC Core, with ~500 LoC, and requested for review, but that didn't get reviewed/merged as everyone was busy.

So a week later I bumped the branch to another branch with GC Repacking & ADR, which were originally intended to be a follow up PR. And re-requested review.

Eventually the rest (DataModel & Dull System Tests) integration part of the GC was added in for the same reason a week after that. I didn't want to make a long chain of unmerged PRs that depend on each other; but getting it reviewed either wasn't super pressing, since the new code didn't depend on anything else in the App. I figured I'd let someone review whenever they felt like it.

Eventually when it came to the UI stuff, I asked in standup if I should make a separate PR or bump the existing one. If you may recall, @erri120 responded to just throw it in the same branch, so that I did.

That's how we got here. The increments were modular, and it's actually just 4 PRs, just that the first one always got bumped instead of making a long chain.

File: tests/NexusMods.DataModel.Synchronizer.Tests/GeneralLoadoutManagementTests.SynchronizerIntegrationTests.verified.md (hunk 1 above)
Comment (halgari) [reply tests/NexusMods.DataModel.Synchronizer.Tests/GeneralLoadoutManagementTests.SynchronizerIntegrationTests.verified.md:117]: Perfect, thanks

**Comment by Al12rs** [comment]
> I did actually try to split it up this time, but no fish bit the line on the review part unfortunately. Originally this PR was #1808 .

I think there was a small miscommunication, there was still work to be done and it seemed like the intention was to add that to the existing PR, which is why it was not reviewed. 

For partial PRs it is good to communicate explicitly that it is ready for review, what is the scope is, as well as what changes are planned or postponed for future PR, so reviewer knows what is expected to be missing and what feedback will be addressed in this PR.

**Pull request merged**